pom.xml

```diff
@@ -118,8 +154,17 @@
                 <plugin>
                     <groupId>org.jenkins-ci.tools</groupId>
                     <artifactId>maven-hpi-plugin</artifactId>
+                    <extensions>true</extensions>
                     <configuration>
-                        <pluginFirstClassLoader>true</pluginFirstClassLoader>
+                        <pluginFirstClassLoader>false</pluginFirstClassLoader>
```

Comment:
Well, if this turns out causing problems the only way to hotfix is reverting the feature as a whole or find way to keep it working with plugin-first classloading in a hurry (mind that it will break existing use-cases not just the new ones). That is why I prefer to find the way to get pipeline working with this setup now.

Let me give it some testing.